### ISSUE TITLE
Any use of pickle, cPickle, marshal or pyyaml where the filename argument can't be trusted

### ISSUE THREAD

**Issue by peterbe** [st3fan/pythoncodeanalysis#3]
Some web frameworks might allow user input or unescaped data from a database that eventually is sent to be unpickled, unmarshalled or opened with pyYaml.

This is all insecure. 

For example, if I see:

``` python
import cPickle
...
cPickle.load(open(filename))
```

I will trace back the variable `filename` careful until I see it's a hardcoded string of sorts and not something a hacker could possibly control.

**Comment by jbremer** [comment]
Yup, this is a good suggestion. Thanks. `pickle` and `cPickle` were already on the list of sinks (i.e., functions which should never be called with data that is possibly provided by the user.) I'll investigate `marshal` and `pyyaml` as well, haven't used these before myself.

**Comment by peterbe** [comment]
`marshal` is just like `pickle` but can be faster. `pyYaml` is not available in the python standard library.
